src/App.tsx

```diff
@@ -1,24 +1,242 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
 import { UserWarning } from './UserWarning';
+import { Header } from './components/Header';
+import { Todo } from './types/Todo';
+import {
+  deleteTodo,
+  getTodos,
+  patchTodo,
+  postTodo,
+} from './api/todos';
+import { TodoList } from './components/TodoList';
+import { TodoFilter } from './components/TodoFilter';
+import { Notification } from './components/Notification';
+import { StatusTodos } from './types/StatusTodo';
 
-const USER_ID = 0;
+const USER_ID = 9946;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [status, setStatus] = useState(StatusTodos.ALL);
+  const [isError, setIsError] = useState(false);
+  const [query, setQuery] = useState('');
+  const [errorMessage, setErrorMessage] = useState('');
+  const [isTodoId, setIsTodoId] = useState<number[] | null>([]);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const isTodoCompleted = todos.every(todo => todo.completed);
+
+  const filteredTodos = useMemo(() => {
+    return todos.filter(todo => {
```

Comment:
You should know you can use an implicit return for the such cases. It's not necessarily but your code will be shorter.

Reply:
pls check my first comments👆🏻